E-Commerce/E-Commerce/Presentation/Views/Home/DetailView/DetailViewController.swift

```diff
@@ -0,0 +1,88 @@
+//
+//  DetailViewController.swift
+//  E-Commerce
+//
+//  Created by Orlando Nicolas Marchioli on 30/01/2024.
+//
+
+import Foundation
+import UIKit
+
+class DetailViewController: UIViewController{
+    
+    @IBOutlet weak var productImage: UIImageView!
+    @IBOutlet weak var productName: UILabel!
+    @IBOutlet weak var productPrice: UILabel!
+    @IBOutlet weak var productQuantity: UILabel!
+    @IBOutlet weak var productDescription: UILabel!
+    @IBOutlet weak var addToCartButton: UIButton!
+    @IBOutlet weak var lessButton: UIButton!
+    @IBOutlet weak var plusButton: UIButton!
+    var delegate: DetailViewControllerDelegate?
+    var product: Product = Product(image: "", name: "", type: "", price: 0)
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        
+        productQuantity.text = "0"
+        plusButton.setImage(UIImage(systemName: "plus"), for: .normal)
+        plusButton.setTitle("", for: .normal)
+        plusButton.imageView?.contentMode = .scaleAspectFit
+        plusButton.layer.cornerRadius = plusButton.bounds.width / 2
+        
+        lessButton.setImage(UIImage(systemName: "minus"), for: .normal)
+        lessButton.setTitle("", for: .normal)
+        lessButton.imageView?.contentMode = .scaleAspectFit
+        lessButton.layer.cornerRadius = lessButton.bounds.width / 2
+        
+        addToCartButton.layer.cornerRadius = 10
+        addToCartButton.isEnabled = false
+        
+        configure(product: product)
+    }
+    @IBAction func addQuantity(_ sender: Any) {
+        self.productQuantity.text = String(Int(self.productQuantity.text!)! + 1)
+        checkAddToChartState()
+    }
+    
+    @IBAction func lessQuantity(_ sender: Any) {
+        if(self.productQuantity.text! != "0"){
+            self.productQuantity.text = String(Int(self.productQuantity.text!)! - 1)
+        }
+        checkAddToChartState()
```

Comment:
I think reduceQuantity() would be a better name. Also force unwrapping (!) this much is a bit dangerous as it could crash the code. I suggest handling this under a guard-let statement to make sure productQuantity.text is not null.